primitives/core/src/traits.rs

```diff
@@ -74,6 +77,15 @@ pub trait BareCryptoStore: Send + Sync {
 	///
 	/// Returns `true` iff all private keys could be found.
 	fn has_keys(&self, public_keys: &[(Vec<u8>, KeyTypeId)]) -> bool;
+
+	/// Signs a message with the private key that matches
+	/// the public key passed.
+	fn sign_with(
+		&self,
+		kind: KeyKindId,
+		id: KeyTypeId,
+		msg: &[u8],
```

Comment:
context missing.

Reply:
As stated elsewhere, this is not the case anymore.